### PR TITLE
Install scripts, bdist_wheel, x86 docker and instructions

### PR DESCRIPTION
**Pull request by passalis**
This PR adds the following:
- [x] Scripts to install OpenDR toolkit in clean Ubuntu 20.04 systems (even when running from a minimal image, e.g,. docker ones)
- [x] `setup.py` to correctly install OpenDR package
- [x] Corrected scripts to activate OpenDR venv environment
- [x] Scripts to create bdist wheels for cpu only usage
- [x] Dockerfile for assembling cpu-only OpenDR inference 
- [x] Readme listing different installation options
- [x] Update wiki to reflect the changes made in this PR

This PR also add missing `__init__.py` in toolkit and a `__version__` variable according to typical python usage.

### CODE REVIEW

**Comment by ad-daniel** [comment]
@passalis I'll look into it.

PS: I think I have been blacklisted, when I try to send a mail to thomas it's refused. Can you or somebody else check?

**Comment by passalis** [comment]
This *was* ready to review... I am trying to fix an dependency issue that was created by detectron dependencies. I will post when I have fixed this issue.

**Comment by passalis** [comment]
I think I fixed the issue by removing detectron from pip wheel dependencies. detectron is a dependency that is needed only by a `project` tool, while the pip wheel currently only includes tools in under `src` so I think is probably ok.

**Review by ad-daniel — CHANGES_REQUESTED** [review]
Thank you! I've left some comments for possible improvements

File: docs/reference/installation.md
Comment (passalis) [reply docs/reference/installation.md:75]: I should also add the missing link to docker here...

File: docs/reference/installation.md
Comment (passalis) [reply docs/reference/installation.md:25]: I think we should go with the later indeed. On my system the first takes 2m9,513s, while the later only 9,287s.

**Comment by passalis** [comment]
Thank you for the suggestions! I will test one more the workflow on a clean install today and I well let you know if I find any additional issue.

**Comment by passalis** [comment]
I have checked the installation instructions on clean Ubuntu 20.04 installation and updated the installation scripts/instructions by adding a few more commands to make sure that there is no dependency missing. As far as I understand both the local/docker install, as well as the generated wheel works on clean systems.

**Comment by passalis** [comment]
I think that I have also managed to provide CUDA dockers. I have built this image after cleaning my local docker cache and it was built successfully.

**Comment by ad-daniel** [comment]
In the current version (CPU only) I'm getting this:
`error in OpenDR setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Invalid URL given`
the requirements.txt file seems correctly formatted, so it might be just a matter that it isn't loaded correctly

**Comment by passalis** [comment]
This is the output of the `./bin/install.sh`?

**Comment by ad-daniel** [comment]
Yes, I've also replaced the `install_requires = fp.read()` with `install_requires = fp.read().splitlines()` but I get the same thing. It might be the gihub references it doesn't like, I'm guessing.

**Comment by passalis** [comment]
It should work as it is... Can you try updating your setuptools, `pip install setuptools -U` and let me know if this helps.

**Comment by ad-daniel** [comment]
I'm having the same issue after upgrading and with splitlines. Let me re-check on a clean system

**Comment by passalis** [comment]
I suspect that this is caused either by pip or setuptools. You can also try updating pip as well.

You can let me know the version of pip and setuptools you have to try to see if I can replicate this on my machine. Also, please make sure that you have activated opendr vev since I think `install.sh` automatically deactivates the environment, even when a command fails.

**Comment by ad-daniel** [comment]
Just a heads up for everybody, when following the 1st install procedure, you can't do a shallow clone as suggested by the guide since currently these changes are in the `install_scripts` branch only.

So instead of doing:
```
git clone --depth 1 --recurse-submodules -j8 https://github.com/opendr-eu/opendr`
```
You need to do:
```
git clone --recurse-submodules -j8 https://github.com/opendr-eu/opendr
cd opendr
git checkout install_scripts
```

**Review by vniclas — COMMENTED** [review]
Disclaimer: The general feedback and method 1) are based on tests performed yesterday before the new commits. I do not have enough time before the deadline to check whether that is still applicable.


General feedback:

1. Can we rename the `master` branch to `main` following the current trend (and GitHub's new defaults). I think this would send a good signal given that we are a publicly funded project.

2. The repo is huge!. This is solely caused by the `.git/objects` folder (2.4GB).

Installation:
1) cloning the repo

Error when installing the GPU version:
- `error in OpenDR setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Invalid URL given`

CPU version:
This generally works, but:
- In the first attempt, the installation crashed my computer --> Add a disclaimer that this requires lots of RAM and one should closer other stuff (10GB should be sufficient though)
- Installing `python-is-python2` is not very user-friendly. Can we remove this or at least clearly list the APT dependencies in the installation procedure. So that the user is aware of these system-wide packages.
- I did not test with CUDA 10.2 (have other version installed and the instructions are system wide instructions).

There are still package inconsistencies and silent overwrites, e.g., 
- detectron2 0.5+cpu requires Pillow>=7.1, but you have pillow 7.0.0 which is incompatible.
- detectron2 0.5+cpu requires pycocotools>=2.0.2, but you have pycocotools 2.0 which is incompatible.


2) pip
The installation succeeds. What is the best way to actually test it? Can we run unittests?

3) CPU docker
Where to get the docker file from?

4) GPU docker
I tried following the installation procedure twice, starting with the installation of nvidia-docker. Both times, docker did not work anymore (running the test `sudo docker run --rm --gpus all nvidia/cuda:11.0-base nvidia-smi`) after adapting the `/etc/docker/daemon.json` as instructed in the `installation.md`.
Unfortunately, I do not have time to further go down this rabbit hole.

File: docs/reference/installation.md
Comment (jelledouwe) [inline docs/reference/installation.md:88]: Perhaps we could also mention that you should run this command in the folder where the docker is located?

File: docs/reference/installation.md
Comment (ad-daniel) [reply docs/reference/installation.md:28]: It's slow now since we need access to the install_scripts branch so we need to clone more than just the master branch. Once it's merged and cloning with --depth 1 it should be almost instantaneous (~10sec).

Comment (vniclas) [reply docs/reference/installation.md:28]: I was referring to the installation.

**Review by jelledouwe — CHANGES_REQUESTED** [review]
All three installation procedure work nicely for me, although I could not test the gpu-enabled versions. Furthermore, I have found two small typos and have some minor suggestions related to documentation. Also I have some remarks, I am not sure whether these need to be adressed, but I am stating them here for completeness sake. For the docker, after running ```bash sudo docker run -p 8888:8888 opendr/ubuntu```, there are links to the Notebook shown in the terminal but not all of them seem to be working. Also, there were a number of warnings issued during installation of the docker.

File: docs/reference/installation.md
Comment (vniclas) [reply docs/reference/installation.md:88]: That's a good note. Probably, more detailed instructions would have also made it work on my machine.

Comment (iliiliiliili) [reply docs/reference/installation.md:88]: As I understood, the docker build should work without cloning the repo manually, which means that the user only needs a link to a Dockerfile (which is in the root folder now), so we should add the link to it here in order to make it easier to download it and start the docker.

Comment (ad-daniel) [reply docs/reference/installation.md:88]: I think the image, once built, will be uploaded to dockerhub so that it can be run with a `docker run -it opendr-eu/opendr-toolkit:latest`. These instructions are more about building it locally

Comment (iliiliiliili) [reply docs/reference/installation.md:88]: @ad-daniel In this case, should we have instructions for two ways to use Docker, locally and via dockerhub?

**Comment by iliiliiliili** [comment]
I used a workstation with Ubuntu 18 on it to test installations.
Pip and clone installations are working ok.
For the docker installation, it was a bit confusing about where to find Dockerfile, so it would be good to have a link for it (even if we will have a dockerhub image).
For the CUDA docker, it failed for me. I installed the nvidia-docker and tested it:
```
sudo docker run --rm --gpus all nvidia/cuda:11.0-base nvidia-smi
Unable to find image 'nvidia/cuda:11.0-base' locally
11.0-base: Pulling from nvidia/cuda
54ee1f796a1e: Pull complete                                                                                                                                                                                                                  f7bfea53ad12: Pull complete                                                                                                                                                                                                                  46d371e02073: Pull complete                                                                                                                                                                                                                  b66c17bbf772: Pull complete                                                                                                                                                                                                                  3642f1a6dfb3: Pull complete                                                                                                                                                                                                                  e5ce55b8b4b9: Pull complete                                                                                                                                                                                                                  155bc0332b0a: Pull complete                                                                                                                                                                                                                  Digest: sha256:774ca3d612de15213102c2dbbba55df44dc5cf9870ca2be6c6e9c627fa63d67a
Status: Downloaded newer image for nvidia/cuda:11.0-base
Tue Dec 21 10:45:55 2021
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 495.29.05    Driver Version: 495.29.05    CUDA Version: 11.5     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  On   | 00000000:02:00.0 Off |                  N/A |
| 25%   30C    P8    10W / 250W |      1MiB / 11178MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce ...  On   | 00000000:03:00.0 Off |                  N/A |
| 25%   29C    P8    11W / 250W |      1MiB / 11178MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  NVIDIA GeForce ...  On   | 00000000:82:00.0 Off |                  N/A |
| 25%   33C    P8    11W / 250W |      1MiB / 11178MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   3  NVIDIA GeForce ...  On   | 00000000:83:00.0 Off |                  N/A |
| 25%   32C    P8    12W / 250W |      1MiB / 11178MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
``` 

Then I updated the `/etc/docker/daemon.json` and after `sudo systemctl restart docker.service` I got this error:
```
Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.
```
Running `journalctl -xe`:
```
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- Unit docker.service has failed.
--
-- The result is RESULT.
Dec 21 10:46:41 5125-126-001.uni.au.dk systemd[1]: docker.service: Service hold-off time over, scheduling restart.
Dec 21 10:46:41 5125-126-001.uni.au.dk systemd[1]: docker.service: Scheduled restart job, restart counter is at 2.
-- Subject: Automatic restarting of a unit has been scheduled
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- Automatic restarting of the unit docker.service has been scheduled, as the result for
-- the configured Restart= setting for the unit.
Dec 21 10:46:41 5125-126-001.uni.au.dk systemd[1]: Stopped Docker Application Container Engine.
-- Subject: Unit docker.service has finished shutting down
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- Unit docker.service has finished shutting down.
Dec 21 10:46:41 5125-126-001.uni.au.dk systemd[1]: Starting Docker Application Container Engine...
-- Subject: Unit docker.service has begun start-up
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- Unit docker.service has begun starting up.
Dec 21 10:46:41 5125-126-001.uni.au.dk dockerd[20164]: unable to configure the Docker daemon with file /etc/docker/daemon.json: configuration validation from file failed: specified default runtime 'nvidia-container-runtime' does not exis
Dec 21 10:46:41 5125-126-001.uni.au.dk systemd[1]: docker.service: Main process exited, code=exited, status=1/FAILURE
Dec 21 10:46:41 5125-126-001.uni.au.dk systemd[1]: docker.service: Failed with result 'exit-code'.
Dec 21 10:46:41 5125-126-001.uni.au.dk systemd[1]: Failed to start Docker Application Container Engine.
-- Subject: Unit docker.service has failed
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- Unit docker.service has failed.
--
-- The result is RESULT.
Dec 21 10:46:43 5125-126-001.uni.au.dk systemd[1]: docker.service: Service hold-off time over, scheduling restart.
Dec 21 10:46:43 5125-126-001.uni.au.dk systemd[1]: docker.service: Scheduled restart job, restart counter is at 3.
-- Subject: Automatic restarting of a unit has been scheduled
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- Automatic restarting of the unit docker.service has been scheduled, as the result for
-- the configured Restart= setting for the unit.
Dec 21 10:46:43 5125-126-001.uni.au.dk systemd[1]: Stopped Docker Application Container Engine.
-- Subject: Unit docker.service has finished shutting down
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- Unit docker.service has finished shutting down.
Dec 21 10:46:43 5125-126-001.uni.au.dk systemd[1]: docker.service: Start request repeated too quickly.
Dec 21 10:46:43 5125-126-001.uni.au.dk systemd[1]: docker.service: Failed with result 'exit-code'.
Dec 21 10:46:43 5125-126-001.uni.au.dk systemd[1]: Failed to start Docker Application Container Engine.
-- Subject: Unit docker.service has failed
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- Unit docker.service has failed.
--
-- The result is RESULT.
Dec 21 10:46:43 5125-126-001.uni.au.dk systemd[1]: docker.socket: Failed with result 'service-start-limit-hit'.
```

We can see that it says
```
Dec 21 10:46:41 5125-126-001.uni.au.dk dockerd[20164]: unable to configure the Docker daemon with file /etc/docker/daemon.json: configuration validation from file failed: specified default runtime 'nvidia-container-runtime' does not exis
```
Nvidia docker works, but the `nvidia-container-runtime` is not found. Should it be something else?

**Comment by katerynaCh** [comment]
Installing from pip and by cloning seem to work fine for me on Ubuntu 20.04. I was not able to test the docker unfortunately.

**Comment by passalis** [comment]
> Disclaimer: The general feedback and method 1) are based on tests performed yesterday before the new commits. I do not have enough time before the deadline to check whether that is still applicable.
> 
> General feedback:
> 
> 1. Can we rename the `master` branch to `main` following the current trend (and GitHub's new defaults). I think this would send a good signal given that we are a publicly funded project.
> 2. The repo is huge!. This is solely caused by the `.git/objects` folder (2.4GB).

We will look into this. Cloning using the command provided in the instructions is much more efficient.

> Installation:
> 
> 1. cloning the repo
> 
> Error when installing the GPU version:
> 
> * `error in OpenDR setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Invalid URL given`
> 

@ad-daniel also reported a similar issue. I will further look into this and come back.

> CPU version: This generally works, but:
> 
> * In the first attempt, the installation crashed my computer --> Add a disclaimer that this requires lots of RAM and one should close other stuff (10GB should be sufficient though)
> * Installing `python-is-python2` is not very user-friendly. Can we remove this or at least clearly list the APT dependencies in the installation procedure. So that the user is aware of these system-wide packages.

I added a disclaimer in the instructions for both. I will check if I can remove the `python` dependency from the installations script (I guess we should be able to do this). 


> * I did not test with CUDA 10.2 (have another version installed and the instructions are system-wide instructions).
> 
> There are still package inconsistencies and silent overwrites, e.g.,
> 
> * detectron2 0.5+cpu requires Pillow>=7.1, but you have pillow 7.0.0 which is incompatible.
> * detectron2 0.5+cpu requires pycocotools>=2.0.2, but you have pycocotools 2.0 which is incompatible.

At this point I am not sure what we can do for these at this point. I will check if we can update, but the last time I checked, this created other issues.

> 2. pip
>    The installation succeeds. What is the best way to actually test it? Can we run unit tests?
You can try running some tests/demos. Only the Python API is contained in the wheel.

> 3. CPU docker
>    Where to get the docker file from?
Added some additional instructions.

> 4. GPU docker
>    I tried following the installation procedure twice, starting with the installation of nvidia-docker. Both times, docker did not work anymore (running the test `sudo docker run --rm --gpus all nvidia/cuda:11.0-base nvidia-smi`) after adapting the `/etc/docker/daemon.json` as instructed in the `installation.md`.
>    Unfortunately, I do not have time to further go down this rabbit hole.
I will check again and we will also provide pre-built images for this as well.

File: README.md
Comment (passalis) [reply README.md:58]: I think the text is in US English (e.g., optimized).

File: setup.py
Comment (passalis) [reply setup.py:46]: As far as I know, no at this point. However, we can change this at a later stage, since CYB will be the toolkit maintainer.

File: docs/reference/installation.md
Comment (passalis) [reply docs/reference/installation.md:28]: Added a note that explains that this will take some time.

File: docs/reference/installation.md
Comment (passalis) [reply docs/reference/installation.md]: Fixed!

File: README.md
Comment (vniclas) [reply README.md:58]: Then, it's inconsistent, e.g., _categorise_ instead of _categorize_.

**Comment by passalis** [comment]
> For the docker, after running `bash sudo docker run -p 8888:8888 opendr/ubuntu`, there are links to the Notebook shown in the terminal but not all of them seem to be working.

@jelledouwe did you tried using the localhost link? I think that are indeed two links and only the `localhost` link works. I will check again and revise the instructions accordingly.

File: README.md
Comment (passalis) [reply README.md:58]: Indeed :P I fixed this instead.

File: docs/reference/installation.md
Comment (passalis) [reply docs/reference/installation.md]: I updated the instructions. I will add the commands for pulling the docker images later today.

File: docs/reference/installation.md
Comment (passalis) [reply docs/reference/installation.md]: This is true. I changed this to "inference/training device"

**Comment by passalis** [comment]
> For the docker installation, it was a bit confusing about where to find Dockerfile, so it would be good to have a link for it (even if we will have a dockerhub image).

I will add links to the dockerfiles in the instructions.



> Nvidia docker works, but the `nvidia-container-runtime` is not found. Should it be something else?
> After installation without the `/etc/docker/daemon.json`, it seems to work and the Jupyter lab has access to all 4 GPUs on the workstation and can create tensors on them.

Actually, this is only needed to make sure that nvidia runtime is available when building the docker image. I will build this images and push them to the dockerhub and then update accordingly.

**Comment by passalis** [comment]
> * error in OpenDR setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Invalid URL given

@vniclas would it be possible to send to me or post the contents of `requirements.txt`? This bug seems to affect at least two people and I cannot understand how to replicate this.

**Comment by ad-daniel** [comment]
For what is worth I'm not having that issue any longer personally, so I'm not sure what triggers it

**Comment by jelledouwe** [comment]
> > For the docker, after running `bash sudo docker run -p 8888:8888 opendr/ubuntu`, there are links to the Notebook shown in the terminal but not all of them seem to be working.
> 
> @jelledouwe did you tried using the localhost link? I think that are indeed two links and only the `localhost` link works. I will check again and revise the instructions accordingly.

@passalis for me the one starting with http://127.0.0.1:8888/ is working and the others are not

**Comment by vniclas** [comment]
> > * error in OpenDR setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Invalid URL given
> 
> @vniclas would it be possible to send to me or post the contents of `requirements.txt`? This bug seems to affect at least two people and I cannot understand how to replicate this.

@passalis One of the commits since yesterday most likely fixed it. I cannot longer reproduce it either.

**Comment by passalis** [comment]
That's sounds very good :-D

I have also build and pushed the cpu images for docker. We will update them with the latest release when the toolkit is ready, but you should probably be able to directly use them by:
```
sudo docker run -p 8888:8888 opendr/opendr-toolkit:cpu_latest
```

**Comment by ad-daniel** [comment]
Just tested the docker and indeed it appears to be working nicely. However connecting to it using simply 127.0.0.1:8888 doesn't as it requests a login. So you need use the provided link `http://127.0.0.1:8888/?token=SOMETHING`

**Comment by iliiliiliili** [comment]
@ad-daniel When you start the Jupyter lab, it shows the link that you can follow, which contains the token value in it.

**Comment by passalis** [comment]
Cuda instructions for docker turned to have a typo. Default should be `"default-runtime": "nvidia` instead of `"default-runtime": "nvidia-default-runtime`. Now, I am testing again and building on a clean system.

**Comment by passalis** [comment]
CUDA docker also appears to work (I have also push an early version to dockerhub). I have simplified a bit the dockerfile in a attempt to reduce the size of the resulting image.

**Comment by passalis** [comment]
I am still compiling the latest Dockerfile-cuda, but this will probably take some time. However, I didn't change anything important, so I hope that the latest change will not broke it. The `opendr-toolkit:cuda` image is also available on Dockerhub. 

@vniclas @jelledouwe @iliiliiliili @katerynaCh could you please check quickly that I didn't miss any of your comments for this PR and if the latest changes fixed all of your issues (I hope that the rest of toolkit still works :smiley: )? Thank you once again for your valuable help!

**Comment by ad-daniel** [comment]
The cuda docker doesn't appear to find my graphic card on its own, it required adding `--gpus all` to the docker run command for it to work. It's probably safer to add it anyway so I've changed the instructions (in D7.2).

After doing this change, the 4 demos I've tested (that rely on cuda) all worked nicely.

**Comment by jelledouwe** [comment]
> I am still compiling the latest Dockerfile-cuda, but this will probably take some time. However, I didn't change anything important, so I hope that the latest change will not broke it. The `opendr-toolkit:cuda` image is also available on Dockerhub.
> 
> @vniclas @jelledouwe @iliiliiliili @katerynaCh could you please check quickly that I didn't miss any of your comments for this PR and if the latest changes fixed all of your issues (I hope that the rest of toolkit still works smiley )? Thank you once again for your valuable help!

Mine are included, thank you @passalis!

**Comment by passalis** [comment]
I think `bcolz` was not built correctly in gpu docker image. I am checking this right now.

**Comment by vniclas** [comment]
Hi @passalis, the majority of my comments have been addressed.

Concerning 'requires a lot of RAM', I would rephrase the current sentence `make sure that you have enough RAM available` with a guess of how much will be actually required.

With respect to the pip packages, `detectron2` does not actually trigger overwriting the installed versions, i.e., as long as `detectron2` still works as expected, this probably does not matter.

Another thing that came to my mind is that it would probably be good to add brief instructions for the user on how to run unittests to verify the successful installation of the toolkit.

**Comment by passalis** [comment]
> I think `bcolz` was not built correctly in gpu docker image. I am checking this right now.

Actually, it works ok. I was testing a previous image. The image currently uploaded on dockerhub seems to work ok.

**Comment by passalis** [comment]
@jelledouwe @vniclas  thanks for your response!

> Concerning 'requires a lot of RAM', I would rephrase the current sentence `make sure that you have enough RAM available` with a guess of how much will be actually required.

I will run some tests now and measure memory consumption.

> Another thing that came to my mind is that it would probably be good to add brief instructions for the user on how to run unittests to verify the successful installation of the toolkit.

I will add some instructions for running the tests to verify the installation.

**Comment by passalis** [comment]
I added a RAM estimate for installation and updated installation instructions for pip wheel, as well as for running the tests.

**Comment by ad-daniel** [comment]
I've moved the instructions to build your own docker image to its own section to avoid confusion.
Also, before merging this PR we need to remember to change all mentions of the `install_scripts` branch

**Comment by passalis** [comment]
I added some instructions for connecting the local X server into the docker (e.g., for displaying OpenCV windows). Also, I removed all references to `install_scripts` so some commands/instructions are now broken. As soon as we merge this PR, I will test everything again to make sure that they work as intended.

**Review by ad-daniel — APPROVED** [review]
For it to work, personally I had to run `xhost +local:root` on my host machine. Otherwise I was getting `qt.qpa.xcb: could not connect to display unix:1
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/opendr/venv/lib/python3.8/site-packages/cv2/qt/plugins" even though it was found.` when trying to run detr's `inference_demo`. But could be just me.

Other than that, the PR looks good to me.

@jelledouwe if your points have been addressed could you approve the PR or we won't be able to merge it if you have pending request changes (and also we need 2 approvals)

**Comment by passalis** [comment]
You are right @ad-daniel I also had to run this command, but I though it was not needed eventually. So I revised the instructions.

**Review by ad-daniel — APPROVED** [review]
Thank you!

**Review by Pavlos-Tosidis — APPROVED** [review]

**Pull request merged**
